### DISCUSSION TITLE
override linker,Could not resolve reference to xxx  at langium 3.0

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1546] [Q&A]
i have a  outer library function called `ADD`, by default ,if i refer it at my st file ,it report
![image](https://github.com/eclipse-langium/langium/assets/26995040/3977625a-eb1f-451c-be34-6b73161dcdd1)
so i  consider override DefaultLinker$getCandidate to return my custom AstNodeDescription instead of LinkingError,
it works well in langium 1.0 and didn't report ` Could not resolve reference to StFunction named 'ADD' ` error

but have some problem in langium 3.0,if i write ADD() without saving(ctrl+s),it work out well but if i press ctrl+s,it stilll report
`Could not resolve reference to StFunction named 'ADD' `
or another situation,i press ctrl+s,it report this error,but restart vscode extension can also work out well

**Comment by zrwsmd** [comment]
![image](https://github.com/eclipse-langium/langium/assets/26995040/4fc44909-0d66-46b6-bfb5-3b6ef42568e9)

**Comment by msujew** [comment]
Can you give a reproducible example? Without it, it's impossible to tell what goes wrong.

**Reply by zrwsmd** [reply]
ok, i will try to provider a example next week,tomorrow is weekend
